src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/custom.py

```diff
@@ -32,6 +34,33 @@ def show_cloud(cloud_name=None):
         raise CLIError(e)
 
 
+def _populate_from_metadata_endpoint(cloud, arm_endpoint):
+    endpoints_in_metadata = ['gallery', 'active_directory_graph_resource_id',
+                             'active_directory_resource_id', 'active_directory']
+    if not arm_endpoint or all([cloud.endpoints.has_endpoint_set(n) for n in endpoints_in_metadata]):
+        return
+    try:
+        import requests
+        metadata_endpoint = arm_endpoint + METADATA_ENDPOINT_SUFFIX
+        response = requests.get(metadata_endpoint)
```

Comment:
let us be consistent and make it work with proxy or cert bundle. https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/_debug.py#L18